game_event/tests.py

```diff
@@ -286,3 +289,41 @@ def test_fail_add_player_to_event_already_in_it(self, client, saved_game_event,
         assert response.status_code == 200
         with pytest.raises(IntegrityError):
             response = client.post(url, {'answer': 'yes'})
+
+
+@pytest.mark.django_db
+class TestWeatherApi:
```

Comment:
@OhadSeg 
We shouldn't.
This test case is kind of testing the library rather than the code.

Also, unit tests should not try and interact with external APIs.

The way to go about it is to use the concept of mocks: https://docs.python.org/3/library/unittest.mock.html
It allows you to inject an object that can be used by the code as if it was that library, but in reality, it will return whatever you tell it to return.

You should have a function that does only the call to the API, and you test it by mocking the external library - on each test case you'll mock a different output and verify that the function behaves accordingly.

Then when unit testing functions that use this function, you can mock the function itself rather than the API, so it returns what you need it to return.

It's a bit big. If you want, you can schedule a call and we can look into it together.

Reply:
After a call with @yftacherzog, @IdanZimi and I removed the explicitly stated API Key, and change it to `'appid': os.getenv("APPID")`, so that when running the server, we should attach the API-KEY ad environment variable, like this : 
`APPID='api-key' pipenv run python manage.py runserver 0.0.0.0:8000`.
we also changed the view function, so if someone run the server without attaching the api key, `pipenv run python manage.py runserver 0.0.0.0:8000` there won't be any problems, just that the game events table, will render without it.

Reply:
The test file has conflicts. You'd need to rebase and resolve those.

Reply:
@yftacherzog Done.